odoo/fields.py

```diff
@@ -1536,11 +1537,14 @@ class _String(Field):
     translate = False                   # whether the field is translated
     prefetch = None
     unaccent = True
+    pattern = (None, None, None)        # tuple (description, regex) to validate an input
```

Comment:
I prefer to use `None` as the default for such an attribute, like in `Float._digits`, which is also a tuple.
You can then use the idiomatic expression `self.pattern` to determine whether there is a pattern tuple.

Reply:
Just an idea, what about removing the pattern description, and require the developer to provide some non-empty `help` string?  That makes the error message a bit more sloppy, but would simplify translations and the API itself:
```python
email = fields.Char(pattern=r"@", pattern_flags=re.I, help="The email address must contain character '@'.")
```

Reply:
> I prefer to use None as the default for such an attribute, like in `Float._digits`

The linter was not happy https://runbot.odoo.com/runbot/build/11048036

> Just an idea, what about removing the pattern description, and require the developer to provide some non-empty `help` string?

I like it

Reply:
> The linter was not happy https://runbot.odoo.com/runbot/build/11048036

I share the opinion of the linter. I didn't raised the concern during my review because it is a bit of nitpicking but I prefer (None, None, None). Doing None[0] would break with a cryptic exception and would be a pain to understand.